c7n/resources/secretsmanager.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from c7n.manager import resources
+from c7n.filters import FilterRegistry
+from c7n.query import QueryResourceManager
+from c7n.tags import RemoveTag, Tag, TagActionFilter, TagDelayedAction
+from c7n.utils import local_session
+
+
+@resources.register('secrets-manager')
+class SecretsManager(QueryResourceManager):
+    filters = FilterRegistry('secrets-manager.filters')
+    filters.register('marked-for-op', TagActionFilter)
+    filter_registry = filters
+    permissions = ('secretsmanager:ListSecretVersionIds',)
+
+    class resource_type(object):
+        service = 'secretsmanager'
+        enum_spec = ('list_secrets', 'SecretList', None)
+        detail_spec = ('describe_secret', 'SecretId', 'ARN', None)
+        id = 'ARN'
+        name = 'Name'
+        dimension = None
+        filter_name = None
+
+    def augment(self, resources):
+        client = local_session(self.session_factory).client('secretsmanager')
+
+        def _augment(r):
+            r['Versions'] = client.list_secret_version_ids(SecretId=r['ARN']).get('Versions')
```

Comment:
do we need the versions by default?

Reply:
So the only way to get created/modified dates are by grabbing versions explicitly. At this point I'm leaning more towards a dedicated filter to handle this as secrets can have a large number of versions, especially if users are practicing secret rotation frequently. Plus, there needs to be some additional work to be done regarding handling the paginated results from AWS.